autosklearn/pipeline/components/data_preprocessing/text_encoding/tfidf_encoding.py

```diff
@@ -0,0 +1,114 @@
+from typing import Dict, Optional, Tuple, Union
+
+from ConfigSpace.configuration_space import ConfigurationSpace
+import ConfigSpace.hyperparameters as CSH
+from ConfigSpace import EqualsCondition
+
+import numpy as np
+import pandas as pd
+import itertools
+
+from autosklearn.pipeline.base import DATASET_PROPERTIES_TYPE, PIPELINE_DATA_DTYPE
+from autosklearn.pipeline.components.base import AutoSklearnPreprocessingAlgorithm
+from autosklearn.pipeline.constants import DENSE, SPARSE, UNSIGNED_DATA, INPUT
+
+from sklearn.feature_extraction.text import TfidfVectorizer
+
+
+class TfidfEncoder(AutoSklearnPreprocessingAlgorithm):
+    def __init__(
+        self,
+        ngram_range: Optional[int] = None,
+        use_idf: Optional[bool] = None,
+        min_df_choice: Optional[str] = None,
+        min_df_absolute: Optional[int] = None,
+        min_df_relative: Optional[float] = None,
+        random_state: Optional[Union[int, np.random.RandomState]] = None
+    ) -> None:
+        self.ngram_range = ngram_range
+        self.random_state = random_state
+        self.use_idf = use_idf
+        self.min_df_choice = min_df_choice
+        self.min_df_absolute = min_df_absolute
+        self.min_df_relative = min_df_relative
+
+    def fit(self, X: PIPELINE_DATA_DTYPE, y: Optional[PIPELINE_DATA_DTYPE] = None
+            ) -> 'TfidfEncoder':
+
+        if isinstance(X, pd.DataFrame):
+            # define a CountVectorizer for every feature (implicitly defined by order of columns,
+            # maybe change the list
+            # to a dictionary with features as keys)
+            if self.min_df_choice == "min_df_absolute":
+                self.preprocessor = TfidfVectorizer(min_df=self.min_df_absolute,
+                                                    use_idf=self.use_idf,
+                                                    ngram_range=(1, self.ngram_range))
+            elif self.min_df_choice == "min_df_relative":
+                self.preprocessor = TfidfVectorizer(min_df=self.min_df_relative,
+                                                    use_idf=self.use_idf,
+                                                    ngram_range=(1, self.ngram_range))
+            else:
+                raise KeyError()
+
+            all_text = itertools.chain.from_iterable(X[col] for col in X.columns)
```

Comment:
Same thing here with nans

```suggestion
            all_text = itertools.chain.from_iterable(X[col].dropna() for col in X.columns)
```